### PR TITLE
feat: make scheduler tests use conftest fixtures and add scheduler stop

### PR DESCRIPTION
**Pull request by JonathanGriffe**

### CODE REVIEW

File: tests/test_scheduler.py [hunk 1]
```diff
@@ -59,97 +66,154 @@ def test_simple_interval_scheduler(start_scheduler, start_cli):
 
 
 @pytest.mark.skipif(os.getenv("CI") == "true", reason="test skipped on CI")
-def test_multiple_schedulers(start_scheduler, start_cli):
-    from tests.scheduler_configs.simple_1 import broker, write_loaded_at
+def test_multiple_schedulers(stub_broker, stub_worker, schedulers):
```
Comment (antoinerabany) [inline tests/test_scheduler.py:69]: maybe you don't need to declare `schedulers` as a fixture as it's used only by this test.

File: tests/test_scheduler.py [hunk 2]
```diff
@@ -59,97 +66,154 @@ def test_simple_interval_scheduler(start_scheduler, start_cli):
 
 
 @pytest.mark.skipif(os.getenv("CI") == "true", reason="test skipped on CI")
```
Comment (antoinerabany) [inline tests/test_scheduler.py:68]: I think you can safely remove all these skip on CI as test should work now with your changes

**Review by antoinerabany — CHANGES_REQUESTED** [review]

File: remoulade/scheduler/scheduler.py [hunk 3]
```diff
@@ -108,6 +108,9 @@ def decode(cls, data: bytes) -> "ScheduledJob":
 
 
 class Scheduler:
+
+    stopped = True
```
Comment (antoinerabany) [inline remoulade/scheduler/scheduler.py:112]: Why put it here, I think it should be an instance variable and not a class variable

File: tests/conftest.py [hunk 4]
```diff
@@ -279,15 +295,35 @@ def frozen_datetime():
...
 
 
+@pytest.fixture
+def scheduler(stub_broker):
+    scheduler = new_scheduler(stub_broker)
+    check_redis(scheduler.client)
+    return scheduler
+
+
+def schedulers(n, schedule):
```
Comment (antoinerabany) [inline tests/conftest.py:312]: i think you can put this in the associated test if it's only used once

File: tests/test_scheduler.py [hunk 5]
```diff
@@ -1,155 +1,216 @@
...
 
-    time.sleep(10)
+    scheduler.schedule = [
+        ScheduledJob(
+            actor_name="write_loaded_at",
+            daily_time=(datetime.datetime.utcnow() - datetime.timedelta(seconds=1)).time(),
+        )
+    ]
+    scheduler_thread.start()
+    time.sleep(1)
```
Comment (antoinerabany) [inline tests/test_scheduler.py:162]: are all these time sleep really usefull ? because they can really slow down the test suite

**Review by antoinerabany — CHANGES_REQUESTED** [review]

File: tests/test_scheduler.py (hunk 5 above)
Comment (JonathanGriffe) [reply tests/test_scheduler.py:162]: I will try to minimize every delay

**Pull request merged**